src/utils/dataclass.py

```diff
@@ -71,7 +72,7 @@ def wrapper_decorator(*args, **kwargs):
                 raise DecodeToDataclassException(f'Type {type(list_of_elements)} is not supported.')
 
             if not list_of_elements:
-                return list_of_elements
+                return []
```

Comment:
list_of_elements could be a tuple
I propose to left it as it is

Reply:
In this case we have an incorrect typing. The function should return a list and we rely on this.